Android.mk

```diff
@@ -276,7 +276,15 @@ else
 $(warning Skipping build of hybris-updater-script since HYBRIS_BOOT_PART is not specified)
 endif
 
-HYBRIS_COMMON_ANDROID8_TARGETS := verity_signer boot_signer e2fsdroid vendorimage ramdisk libselinux_stubs libsurfaceflinger libsf_compat_layer libhwc2_compat_layer bootctl
+HYBRIS_COMMON_ANDROID8_TARGETS := e2fsdroid ramdisk libselinux_stubs libsurfaceflinger libsf_compat_layer libhwc2_compat_layer
+# Legacy-partitioned devices have no need to build slotselect or vendorimage
+ifeq ($(BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED),true)
+	HYBRIS_COMMON_ANDROID8_TARGETS += vendorimage bootctl
```

Comment:
Would there be a feasible case where `BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED` is not defined, and yet we still need to build `vendorimage` and `bootctl`? I'm mostly querying MTK device porting experience, cc @krnlyng @NotKit

Reply:
The split is a hard requirement for devices launching with with O and will get set automatically if PRODUCT_FULL_TREBLE is set.
```
ifeq ($(PRODUCT_FULL_TREBLE),true)
  BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED ?= true
endif
```
The only other variable is `PRODUCT_COPY_OUT_VENDOR`, but you need to have `BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED` set to not have boot failures.

Maybe @krnlyng or @NotKit prove me wrong, but I don't see any other possible scenario.

Reply:
@mlehtima @sledges I know it was not added in this PR, but is there a particular reason building vendorimage is needed in the first place? I suppose on LineageOS devices at least just pached libraries get overlayed, and vendor should not be modified in theory?